diff --git a/build/moz.configure/compile-checks.configure b/build/moz.configure/compile-checks.configure
index f710e61474..da468310f4 100644
--- a/build/moz.configure/compile-checks.configure
+++ b/build/moz.configure/compile-checks.configure
@@ -395,7 +395,7 @@ def check_and_add_warning_impl(warning, compiler=None, when=None, check=True):
 
 # When changing this version, please check that no warning support have been
 # removed.
-clang_latest_release = dependable("22.1.0")
+clang_latest_release = dependable("22.1.8")  # GeckoW10m: local clang-cl
 
 
 @depends(c_compiler, clang_latest_release, when=moz_automation)
