diff --git a/media/libwebp/sharpyuv/moz.build b/media/libwebp/sharpyuv/moz.build
index 6ae00382d1..a7c043973c 100644
--- a/media/libwebp/sharpyuv/moz.build
+++ b/media/libwebp/sharpyuv/moz.build
@@ -19,7 +19,9 @@ include('/media/libwebp/libwebp-config.mozbuild')
 # and enabled by what src/dsp/cpu.h does. So just removing the defines here may do
 # nothing.
 
-if CONFIG['TARGET_CPU'] == 'arm' and CONFIG['BUILD_ARM_NEON']:
+if CONFIG['TARGET_CPU'] == 'arm' and (
+    CONFIG['BUILD_ARM_NEON'] or CONFIG['OS_TARGET'] == 'WINNT'
+):
     SOURCES += [
         'sharpyuv_neon.c',
     ]
