diff --git a/media/ffvpx/libavutil/avutil.symbols b/media/ffvpx/libavutil/avutil.symbols
index b5d7c8e5d9..c24ea080ab 100644
--- a/media/ffvpx/libavutil/avutil.symbols
+++ b/media/ffvpx/libavutil/avutil.symbols
@@ -312,7 +312,7 @@ avpriv_float_dsp_alloc
 avpriv_report_missing_feature
 avpriv_request_sample
 avpriv_set_systematic_pal2
-#if defined(XP_WIN) && !defined(_ARM64_)
+#if defined(XP_WIN) && !defined(_ARM64_) && !defined(_ARM_)
 avpriv_emms_asm
 #endif
 avpriv_slicethread_create
