diff --git a/memory/mozalloc/mozalloc.cpp b/memory/mozalloc/mozalloc.cpp
index f8e01fe7f4..5be46a0aa1 100644
--- a/memory/mozalloc/mozalloc.cpp
+++ b/memory/mozalloc/mozalloc.cpp
@@ -38,6 +38,7 @@
 #include "mozilla/Likely.h"
 #include "mozilla/mozalloc.h"
 #include "mozilla/mozalloc_oom.h"  // for mozalloc_handle_oom
+#include "GeckoW10mNote.h"
 
 #if defined(MOZ_MEMORY)
 MOZ_MEMORY_API char* strdup_impl(const char*);
@@ -47,6 +48,10 @@ MOZ_MEMORY_API char* strndup_impl(const char*, size_t);
 void* moz_xmalloc(size_t size) {
   void* ptr = malloc_impl(size);
   if (MOZ_UNLIKELY(!ptr && size)) {
+#if defined(GECKO_W10M)
+    gecko_w10m::NoteFailedAllocation("xmalloc", size, 1,
+                                  __builtin_return_address(0));
+#endif
     mozalloc_handle_oom(size);
     return moz_xmalloc(size);
   }
@@ -58,6 +63,10 @@ void* moz_xcalloc(size_t nmemb, size_t size) {
   if (MOZ_UNLIKELY(!ptr && nmemb && size)) {
     mozilla::CheckedInt<size_t> totalSize =
         mozilla::CheckedInt<size_t>(nmemb) * size;
+#if defined(GECKO_W10M)
+    gecko_w10m::NoteFailedAllocation("xcalloc", nmemb, size,
+                                  __builtin_return_address(0));
+#endif
     mozalloc_handle_oom(totalSize.isValid() ? totalSize.value() : SIZE_MAX);
     return moz_xcalloc(nmemb, size);
   }
@@ -67,6 +76,10 @@ void* moz_xcalloc(size_t nmemb, size_t size) {
 void* moz_xrealloc(void* ptr, size_t size) {
   void* newptr = realloc_impl(ptr, size);
   if (MOZ_UNLIKELY(!newptr && size)) {
+#if defined(GECKO_W10M)
+    gecko_w10m::NoteFailedAllocation("xrealloc", size, 1,
+                                  __builtin_return_address(0));
+#endif
     mozalloc_handle_oom(size);
     return moz_xrealloc(ptr, size);
   }
