diff --git a/third_party/zucchini/chromium/build/build_config.h b/third_party/zucchini/chromium/build/build_config.h
index 2484703699..14cc4c86ff 100644
--- a/third_party/zucchini/chromium/build/build_config.h
+++ b/third_party/zucchini/chromium/build/build_config.h
@@ -298,7 +298,8 @@
 #define ARCH_CPU_PPC64 1
 #define ARCH_CPU_64_BITS 1
 #define ARCH_CPU_LITTLE_ENDIAN 1
-#elif defined(__ARMEL__)
+// clang-cl targeting 32-bit ARM Windows defines _M_ARM, not __ARMEL__.
+#elif defined(__ARMEL__) || defined(_M_ARM)
 #define ARCH_CPU_ARM_FAMILY 1
 #define ARCH_CPU_ARMEL 1
 #define ARCH_CPU_32_BITS 1
