diff --git a/third_party/zucchini/chromium/base/time/time_win.cc b/third_party/zucchini/chromium/base/time/time_win.cc
index 2d93834555..6b4f322dba 100644
--- a/third_party/zucchini/chromium/base/time/time_win.cc
+++ b/third_party/zucchini/chromium/base/time/time_win.cc
@@ -670,7 +670,7 @@ ThreadTicks ThreadTicks::GetForThread(
     const PlatformThreadHandle& thread_handle) {
   DCHECK(IsSupported());
 
-#if defined(ARCH_CPU_ARM64)
+#if defined(ARCH_CPU_ARM_FAMILY)
   // QueryThreadCycleTime versus TSCTicksPerSecond doesn't have much relation to
   // actual elapsed time on Windows on Arm, because QueryThreadCycleTime is
   // backed by the actual number of CPU cycles executed, rather than a
@@ -704,7 +704,7 @@ ThreadTicks ThreadTicks::GetForThread(
 
 // static
 bool ThreadTicks::IsSupportedWin() {
-#if defined(ARCH_CPU_ARM64)
+#if defined(ARCH_CPU_ARM_FAMILY)
   // The Arm implementation does not use QueryThreadCycleTime and therefore does
   // not care about the time stamp counter.
   return true;
@@ -715,7 +715,7 @@ bool ThreadTicks::IsSupportedWin() {
 
 // static
 void ThreadTicks::WaitUntilInitializedWin() {
-#if !defined(ARCH_CPU_ARM64)
+#if !defined(ARCH_CPU_ARM_FAMILY)
   while (time_internal::TSCTicksPerSecond() == 0)
     ::Sleep(10);
 #endif
@@ -762,7 +762,7 @@ ABI::Windows::Foundation::TimeSpan TimeDelta::ToWinrtTimeSpan() const {
   return time_span;
 }
 
-#if !defined(ARCH_CPU_ARM64)
+#if !defined(ARCH_CPU_ARM_FAMILY)
 namespace time_internal {
 
 bool HasConstantRateTSC() {
@@ -829,6 +829,6 @@ double TSCTicksPerSecond() {
 }
 
 }  // namespace time_internal
-#endif  // defined(ARCH_CPU_ARM64)
+#endif  // defined(ARCH_CPU_ARM_FAMILY)
 
 }  // namespace base
